src/java/voldemort/client/ClientConfig.java

```diff
@@ -1094,6 +1106,50 @@ public ClientConfig setMaxBootstrapRetries(int maxBootstrapRetries) {
         this.maxBootstrapRetries = maxBootstrapRetries;
         return this;
     }
+    
+    public int getBootstrapRetryWaitTimeSeconds() {
+        return this.bootstrapRetryWaitTimeSeconds;
+    }
+
+    /**
+     * set the wait time in seconds after a failed bootstrap attempt to next
+     * retry
+     * 
+     * @param value
+     *            wait time in seconds
+     * @throws IllegalArgumentException
+     *             If bootstrapWaitTimeInSeconds < 1
+     */
+    public ClientConfig setBootstrapRetryWaitTimeSeconds(int bootstrapWaitTimeInSeconds) {
+        if (bootstrapWaitTimeInSeconds < 1) {
+            throw new IllegalArgumentException("bootstrap retry wait time should be >= 1");
+        }
+        this.bootstrapRetryWaitTimeSeconds = bootstrapWaitTimeInSeconds;
+        return this;
+    }
+
+    public boolean isFetchAllStoresXmlInBootstrap() {
+        return fetchAllStoresXmlInBootstrap;
+    }
+
+    /**
+     * Voldemort servers older than 1.8.1 supported only fetching all stores.xml
+     * 1.8.1 supported fetching individual stores. During bootstrap this
+     * property controls whether to fetch all stores.xml or only the particular
+     * store.
+     * 
+     * This property is defaulted to true, so that newer clients can work with
+     * older servers. If there are 100's of stores and you have servers running
+     * on 1.8.1 or above, it is better to set them to false.
+     * 
+     * @param value
+     *            true, to fetch all stores.xml false, to fetch only current
+     *            store
+     */
+    public ClientConfig setFetchAllStoresXmlInBootstrap(boolean value) {
```

Comment:
Thanks for the great JavaDoc.

Rather than requiring a client config, it'd be great if the client could automatically detect whether the server supports this optimization. Either by trying the optimized path first and falling back on the non-optimized one if the first one fails, or by asking the server for a certain config (we have a generic admin command now that gives visibility into any arbitrary server config, which may be useful).

I don't think this is a blocker. We can leave it like this. I just wanted to mention it because sometimes doing the auto-tuned client does not require that much more code than introducing a manual config.